Add integration tests for MariaDB

mirromutth opened the pull request:
Motivation:

Add integration tests for MariaDB. See also #178 

Modification:

- Add workflows
- Correct `BitSetCodec` in client-preparing (MariaDB cannot select `BIT` by HEX string)
- Correct JSON test cases for MariaDB
- Add `TEXT` integration test to avoid potential bugs due to differences between MySQL and MariaDB
- Correct README about `BIT` and `TEXT`

Result:

MariaDB integration.

jchrys reviewed (approved):
LGTM

Would you mind consolidating it into a single commit before merging? Thank you!

mirromutth committed:
Add integration for MariaDB

- Add MariaDB workflows
- Correct `BitSetCodec` in client-preparing for MariaDB, it cannot select `BIT` by HEX string
- Correct JSON test cases for MariaDB, it responds `TEXT` for `JSON` type
- Add `TEXT` integration test to avoid potential bugs due to differences between MySQL and MariaDB
- Correct README about `BIT` and `TEXT`
